### PR TITLE
Optional override for Vertex custom job name

### PR DESCRIPTION
**Pull request by jeremy-thomas-roc**
Add optional parameter to VertexAICustomJob to name jobs. This helps in monorepo situations where all Vertex jobs will just be {repo-name}-{uuid}, which is not very helpful when running lots of scheduled jobs.

### Checklist

- [ ] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-gcp/issues/new/choose) first.
- [ ] Includes tests or only affects documentation.
- [X] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [ ] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [X] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-gcp/blob/main/CHANGELOG.md)

### CODE REVIEW

**Comment by jeremy-thomas-roc** [comment]
@desertaxle Can I get some feedback on this? I believe the failing tests happened when there was a dependency issue of the prefect side, but I am unable to re-run them to verify.

**Comment by desertaxle** [comment]
@jeremy-thomas-roc I think we could use the existing `name` attribute on the `VertexAICustomJob` job to create more unique names for created jobs. By default, `name` will be the flow run name, which has the added advantage of making it easier to map Vertex AI jobs to flow runs. Let me know if that would work for your use case or it you have any questions on implementation.

**Comment by jeremy-thomas-roc** [comment]
@desertaxle that works for me, implemented that change

**Comment by desertaxle** [comment]
@jeremy-thomas-roc That's true that `name` can be `None` if the block is used outside of a deployment. A simple check to see if `name` is set should work. We do something similar in `prefect-azure`. You can see https://github.com/PrefectHQ/prefect-azure/pull/92 as an example.

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 ### Changed
+ - `job_name` in `VertexAICustomJob` is now based on the flow run name instead of the image repository.
```
Comment (desertaxle) [inline CHANGELOG.md:13]:
```suggestion
 - Use flow run name for name of created custom jobs - [#208](https://github.com/PrefectHQ/prefect-gcp/pull/208)
```

File: prefect_gcp/aiplatform.py [hunk 2]
```diff
@@ -200,17 +199,14 @@ def job_name(self):
         https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform.CustomJob#google_cloud_aiplatform_CustomJob_display_name
         """  # noqa
         try:
-            repo_name = self.image.split("/")[2]  # `gcr.io/<project_name>/<repo>/`"
+            repo_name = self.name or self.image.split("/")[2]
+            return f"{repo_name}-{uuid4().hex}"
```
Comment (desertaxle) [inline prefect_gcp/aiplatform.py:203]:
```suggestion
            base_name = self.name or self.image.split("/")[2]
            return f"{base_name}-{uuid4().hex}"
```

**Review by desertaxle — CHANGES_REQUESTED** [review]
Couple of small nits, but otherwise this is good to go!

**Comment by jeremy-thomas-roc** [comment]
@desertaxle suggested changes implemented

**Pull request merged**